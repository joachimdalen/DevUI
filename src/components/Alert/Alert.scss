.#{$provider-prefix}-alert {
  border-width: $alert-border-width;
  border-style: solid;
  border-radius: $alert-border-radius;
  padding: $alert-padding;
  margin: 2px;
  position: relative;
  display: flex;
  align-items: center;
  width: 100%;
  .fas {
    font-size: $alert-icon-font-size;
  }
  .#{$provider-prefix}-alert-icon-container {
    margin-right: 5px;
  }
  .#{$provider-prefix}-alert-message {
    font-weight: $alert-message-font-weight;
    font-size: $alert-message-font-size;
    margin: 0;
    color: $alert-message-color;
  }
  .#{$provider-prefix}-alert-description {
    margin-top: 5px;
    color: $alert-description-color;
    font-weight: $alert-description-font-weight;
    font-size: $alert-description-font-size;
  }
  .#{$provider-prefix}-alert-close {
    position: absolute;
    top: 5px;
    right: 10px;
    &:hover {
      cursor: pointer;
      color: $alert-close-hover-color;
    }
  }
  &.#{$provider-prefix}-alert-wi {
    display: inline-flex;
    .#{$provider-prefix}-alert-wi-content {
      margin-left: 10px;
    }
  }
  &.#{$provider-prefix}-alert-info {
    border-color: $alert-info-border-color;
    background-color: $alert-info-background;
    color: $alert-info-color;
  }
  &.#{$provider-prefix}-alert-success {
    border-color: $alert-success-border-color;
    background-color: $alert-success-background;
    color: $alert-success-color;
  }
  &.#{$provider-prefix}-alert-warning {
    border-color: $alert-warning-border-color;
    background-color: $alert-warning-background;
    color: $alert-warning-color;
  }
  &.#{$provider-prefix}-alert-danger {
    border-color: $alert-danger-border-color;
    background-color: $alert-danger-background;
    color: $alert-danger-color;
  }
}
.#{$provider-prefix}-alert {
  border-width: $alert-border-width;
  border-style: solid;
  border-radius: $alert-border-radius;
  padding: $alert-padding;
  margin: 2px;
  position: relative;
  display: flex;
  align-items: center;
  width: 100%;
  .fas {
    font-size: $alert-icon-font-size;
  }
  .#{$provider-prefix}-alert-message {
    font-weight: $alert-message-font-weight;
    font-size: $alert-message-font-size;
    margin: 0;
    color: $alert-message-color;
  }
  .#{$provider-prefix}-alert-description {
    margin-top: 5px;
    color: $alert-description-color;
    font-weight: $alert-description-font-weight;
    font-size: $alert-description-font-size;
  }
  .#{$provider-prefix}-alert-close {
    position: absolute;
    top: 5px;
    right: 10px;
    &:hover {
      cursor: pointer;
      color: $alert-close-hover-color;
    }
  }
  &.#{$provider-prefix}-alert-wi {
    display: inline-flex;
    .#{$provider-prefix}-alert-wi-content {
      margin-left: 10px;
    }
  }
  &.#{$provider-prefix}-alert-info {
    border-color: $alert-info-border-color;
    background-color: $alert-info-background;
    color: $alert-info-color;
  }
  &.#{$provider-prefix}-alert-success {
    border-color: $alert-success-border-color;
    background-color: $alert-success-background;
    color: $alert-success-color;
  }
  &.#{$provider-prefix}-alert-warning {
    border-color: $alert-warning-border-color;
    background-color: $alert-warning-background;
    color: $alert-warning-color;
  }
  &.#{$provider-prefix}-alert-danger {
    border-color: $alert-danger-border-color;
    background-color: $alert-danger-background;
    color: $alert-danger-color;
  }
}
