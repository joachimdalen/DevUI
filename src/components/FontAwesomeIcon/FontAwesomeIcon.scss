.#{$provider-prefix}-icon {
  &.fa,
  &.far,
  &.fas {
    font-family: "Font Awesome 5 Free" !important;
  }
  &.fab {
    font-family: "Font Awesome 5 Brands" !important;
  }

  &-mr {
    margin-right: 5px;
  }
  &-ml {
    margin-left: 5px;
  }
}
