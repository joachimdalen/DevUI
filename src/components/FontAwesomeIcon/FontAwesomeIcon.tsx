import * as React from "react";
import cx from "classnames";
export interface FontAwesomeIconProps {
  icon: string;
  animate?: boolean;
  fixedWidth?: boolean;
  className?: string;
  size?: FAISize;
  iconStyle: FAIStyle;
  marginDirection?: FAIMarginDirection;
  animationType?: FAIAnimationType;
  onClick?: () => void;
}
export type FAIMarginDirection = "left" | "right";
export type FAIAnimationType = "spin" | "pulse";
export type FAISize = "normal" | "large" | "xlarge";
export type FAIStyle = "solid" | "regular" | "light" | "brands";

const _typeFromStyle = (style: FAIStyle) => {
  switch (style) {
    case "solid":
      return "fas";
    case "regular":
      return "far";
    case "light":
      return "fal";
    case "brands":
      return "fab";
  }
};
const _sizeClassFromSize = (size: FAISize) => {
  switch (size) {
    case "large":
      return "fa-2x";
    case "normal":
      return "";
    case "xlarge":
      return "fa-3x";
  }
};

const FontAwesomeIcon = ({
  icon,
  marginDirection,
  animate = false,
  animationType,
  size = "normal",
  className = "",
  fixedWidth,
  iconStyle,
  onClick
}: FontAwesomeIconProps) => {
  const animationClass = animationType === "spin" ? "fa-spin" : "fa-pulse";
  const marginClass =
    marginDirection === "left" ? "dui-icon-ml" : "dui-icon-mr";
  const sizeClass = _sizeClassFromSize(size);

  return (
    <i
      onClick={onClick}
      className={cx(
        "dui-icon",
        { "fa-fw": fixedWidth },
        className,
        _typeFromStyle(iconStyle),
        icon,
        sizeClass,
        { [animationClass]: animate },
        { [marginClass]: marginDirection }
      )}
    />
  );
};
export default FontAwesomeIcon;
