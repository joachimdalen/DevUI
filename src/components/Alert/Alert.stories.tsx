import { select } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import * as React from 'react';

import { Alert, AlertVariant } from './Alert';
const typeOptions = {
  info: 'info',
  warning: 'warning',
  success: 'success',
  danger: 'danger'
};
storiesOf('Display Components/Alert', module)
  .add('Default', () => (
    <Alert
      message="This is the message"
      description="This is the description"
      variant={select('Variant', typeOptions, 'info') as AlertVariant}
    />
  ))
  .add('With icon', () => (
    <Alert
      message="This is the message"
      description="This is the description"
      variant={select('Variant', typeOptions, 'info') as AlertVariant}
      withIcon
    />
  ));
