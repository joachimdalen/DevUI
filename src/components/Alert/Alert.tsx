import * as React from "react";
import cx from "classnames";
import FontAwesomeIcon from "../FontAwesomeIcon/FontAwesomeIcon";

export type AlertVariant = "info" | "success" | "warning" | "danger";
export interface AlertProps {
  className?: string;
  variant?: AlertVariant;
  message: string;
  description?: string;
  onClose?: () => void;
  icon?: string;
}
export class Alert extends React.Component<AlertProps> {
  static defaultProps: Partial<AlertProps> = {
    variant: "info"
  };
  _getIconFromVariant = () => {
    const { variant } = this.props;

    switch (variant) {
      case "info":
        return "fa-info-circle";
      case "success":
        return "fa-check-circle";
      case "warning":
        return "fa-exclamation-circle";
      case "danger":
        return "fa-times-circle";
      default:
        return "";
    }
  };
  render() {
    const {
      className,
      variant,
      message,
      description,
      onClose,
      icon
    } = this.props;
    const closable = onClose !== null;
    const baseComponent = (
      <React.Fragment>
        <h1 className={cx("dui-alert-message")}>{message}</h1>
        {description && (
          <p className={cx("dui-alert-description")}>{description}</p>
        )}
        {closable && (
          <span className={cx("dui-alert-close")} onClick={onClose}>
            <FontAwesomeIcon icon="fas fa-times" />
          </span>
        )}
      </React.Fragment>
    );
    const baseClass = cx(
      "dui-alert",
      "dui-alert-wi",
      `dui-alert-${variant}`,
      className
    );
    if (icon) {
      const itemIcon = icon ? `fa-${icon}` : this._getIconFromVariant();
      return (
        <div className={baseClass}>
          <span>
            <FontAwesomeIcon icon={cx("fas", "fa-fw", itemIcon)} />
          </span>
          <div className={cx("dui-alert-wi-content")}>{baseComponent}</div>
        </div>
      );
    }
    return <div className={baseClass}>{baseComponent}</div>;
  }
}
