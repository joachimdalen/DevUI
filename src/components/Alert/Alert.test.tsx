import React from "react";
import renderer from "react-test-renderer";
import { FontAwesomeIcon } from "../FontAwesomeIcon/FontAwesomeIcon";
import { Alert } from "./Alert";

describe("Alert", () => {
  it("matches snapshot", () => {
    const component = renderer.create(<Alert message="Hello" />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("sets the correct details", () => {
    const component = renderer.create(
      <Alert message="Hello" description="HelloDescription" />
    );
    let alertMessage = component.root.findByType("h1");
    let alertDescription = component.root.findByType("p");

    expect(alertMessage.props.children).toBe("Hello");
    expect(alertDescription.props.children).toBe("HelloDescription");
  });
  it("can be closed", () => {
    const onClose = jest.fn();

    const component = renderer.create(
      <Alert message="Hello" onClose={onClose} />
    );

    let closeButton = component.root.findByProps({
      className: "dui-alert-close"
    });
    closeButton.props.onClick();
    expect(onClose).toBeCalled();
  });
  it("has the correct variant", () => {
    const component = renderer.create(
      <Alert message="hello" variant="success" />
    );
    expect(component.root.findAllByType("div")[0].props.className).toContain(
      "dui-alert-success"
    );
  });
  it("should show the default icon", () => {
    const component = renderer.create(
      <Alert message="hello" variant="success" withIcon />
    );
    expect(component.root.findAllByType("span")[0].props.className).toContain(
      "dui-alert-icon-container"
    );
  });
  it("should show the given icon", () => {
    const component = renderer.create(
      <Alert message="hello" variant="success" withIcon icon="times" />
    );
    const iconContainer = component.root.findAllByType("span")[0];
    expect(iconContainer).not.toBeUndefined();
    const icon = iconContainer.findAllByType(FontAwesomeIcon)[0];
    expect(icon).not.toBeUndefined();
    expect(icon.props.icon).toContain("fa-times");
  });
});
