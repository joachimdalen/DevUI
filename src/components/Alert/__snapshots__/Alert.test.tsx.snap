// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert matches snapshot 1`] = `
<div
  className="dui-alert dui-alert-info"
>
  <h1
    className="dui-alert-message"
  >
    Hello
  </h1>
</div>
`;
